src/clients/java/src/main/java/com/tigerbeetle/UInt128.java

```diff
@@ -172,5 +174,60 @@ public static byte[] asBytes(final BigInteger value) {
         return asBytes(bigintLsb.longValueExact(), bigintMsb.longValueExact());
     }
 
+    private static long idLastTimestamp = 0L;
+    private static final byte[] idLastRandom = new byte[80];
+    private static final SecureRandom idSecureRandom = new SecureRandom();
 
+    /**
+     * Generates a Universally Unique Sortable Identifier as 16 bytes of a 128-bit value.
+     *
+     * The ID() function is thread-safe, the bytes returned are formatted in little endian, and the
+     * unsigned 128-bit value always monotonically increasing.
+     *
+     * @throws ArithmeticException if the random monotonic value in the same millisecond overflows.
+     * @return an array of 16 bytes representing an unsigned 128-bit value in little endian.
+     */
+    public static byte[] ID() {
+        long randomLo;
+        short randomHi;
+        long timestamp = System.currentTimeMillis();
+
+        // Only modify the static variables in the synchronized block.
+        synchronized (idSecureRandom) {
+            // Ensure timestamp is monotonic. If it advances forward, also generate a new random.
+            if (timestamp <= idLastTimestamp) {
+                timestamp = idLastTimestamp;
+            } else {
+                idLastTimestamp = timestamp;
+                idSecureRandom.nextBytes(idLastRandom);
+            }
+
+            var random = ByteBuffer.wrap(idLastRandom).order(ByteOrder.nativeOrder());
+            randomLo = random.getLong();
+            randomHi = random.getShort();
+
+            // If randomLo will overflow from increment, then increment randomHi as carry.
+            // If randomHi will overflow on increment, throw error on 80-bit random overflow.
+            if (randomLo == 0xFFFFFFFFFFFFFFFFL) {
+                if (randomHi == 0xffff) {
```

Comment:
Let's standardize hex in upper case.

```suggestion
                if (randomHi == 0xFFFF) {
```